Blackjack.Tests/ControllerShould.cs

```diff
@@ -2,51 +2,56 @@
 using System.Collections.Generic;
 using System.Linq;
 using Moq;
+using Moq.Protected;
 using Xunit;
 
 namespace Blackjack.Tests
 {
     public class ControllerShould
     {
+        Controller _controller;
+        public ControllerShould()
+        {
+            _controller = new Controller();
+        }
         [Fact]
-        public void WhenInitialized_ReturnsADeck()
+        public void HasFullDeck_WhenInitialized()
         {
-            var deck = new Deck(new List<Card>());
-            var controller = new Controller(deck);
-            var numberOfCardsInSuit = controller.Deck.Cards.Count(x => x.Suit == CardSuit.Clubs);
-            var numberOfCardsOfRank = controller.Deck.Cards.Count(x => x.Rank == CardRank.Ace);
+            var cards = _controller.Deck.Cards;
+            var numberOfCardsInSuit = cards.Count(x => x.Suit == CardSuit.Clubs);
+            var numberOfCardsOfRank = cards.Count(x => x.Rank == CardRank.Ace);
 
-            Assert.Equal(52, controller.Deck.Cards.Count);
+            Assert.Equal(52, cards.Count);
             Assert.Equal(13, numberOfCardsInSuit);
             Assert.Equal(4, numberOfCardsOfRank);
         }
 
         [Fact]
-        public void WhenShuffled_DeckHasSameCards() // TODO: how to call through Controller.Play()
+        public void HaveSameCardsInDeck_AfterShuffling() // TODO: how to call through Controller.Play()
         {
             var deck = new Deck(new List<Card>());
-            var controller = new Controller(deck);
-
-            var result = controller.Shuffle();
-            var numberOfCardsInSuit = result.Cards.Count(x => x.Suit == CardSuit.Clubs);
-            var numberOfCardsOfRank = result.Cards.Count(x => x.Rank == CardRank.Ace);
-
+            
+            var result = _controller.Shuffle();
+            var cards = result.Cards;
+            var numberOfCardsInSuit = cards.Count(x => x.Suit == CardSuit.Clubs);
+            var numberOfCardsOfRank = cards.Count(x => x.Rank == CardRank.Ace);
+            
             Assert.True(BlackjackHelper.DecksOfCardsAreEqual(deck, result));
-            Assert.Equal(52, result.Cards.Count);
+            Assert.Equal(52, cards.Count);
             Assert.Equal(13, numberOfCardsInSuit);
             Assert.Equal(4, numberOfCardsOfRank);
         }
 
         [Fact]
-        public void WhenInitialized_GiveEachPlayer2Cards() // TODO: failing - returns a count of 3 for both asserts
+        public void GiveEachPlayer2Cards_AfterDeckShuffled() // TODO: does this violate the law of demeter?

```

Comment:
yes because you are calling the controller, then the player, then the player's hand, then getting the count of the player's hand. Ideally you should only have to call a method that belongs to the controller to obey the Law of Demeter.
This post has good examples to illustrate the concept: https://betterprogramming.pub/demeters-law-don-t-talk-to-strangers-87bb4af11694